pkg/controllers/dnspolicy/dnspolicy_controller.go

```diff
@@ -194,6 +197,11 @@ func (r *DNSPolicyReconciler) deleteResources(ctx context.Context, dnsPolicy *v1
 			return err
 		}
 	}
+	for _, gw := range gatewayDiffObj.GatewaysWithInvalidPolicyRef {
+		if err := r.DeleteTargetBackReference(ctx, client.ObjectKeyFromObject(dnsPolicy), gw.Gateway, DNSPolicyBackRefAnnotation); err != nil {
+			return err
+		}
+	}
```

Comment:
This shouldn't be required, the direct back ref is being removed [here](https://github.com/Kuadrant/multicluster-gateway-controller/pull/349/files#diff-162ceed42032f121fd671b5884a611158c00f31e292b7602e48d5671ca7fe01fR195-R199) when the target still exists.

This should only be getting called once for the policy we are currently deletingResources for. And the polices (plural) annotation gets dealt with here https://github.com/Kuadrant/multicluster-gateway-controller/pull/349/files#diff-162ceed42032f121fd671b5884a611158c00f31e292b7602e48d5671ca7fe01fR207

This btw is the same for all current kuadrant polices see [AuthPolicy](https://github.com/Kuadrant/kuadrant-operator/blob/main/controllers/authpolicy_controller.go#L173-L181) and https://github.com/Kuadrant/kuadrant-operator/blob/main/controllers/ratelimitpolicy_controller.go#L212-L220 which this is based. If there is some issue that this was trying to fix, I'd expect that to be an issue in all of them.

Reply:
yes, this was also my clutching at straws for the e2e tests.